README.md

```diff
@@ -38,24 +38,52 @@ This will return an object, using the `id` property values as keys:
 }
 ```
 
-`id` and `name` are supported by default. If you want to use another property,
+`id` is the default property name to index by. If you want to use another property,
 specify it as an optional second argument:
 
 ```js
-var tree = objectifyArray(people, 'myCustomKeyName')
+var peopleMap = objectifyArray(people, { by: 'myCustomKeyName' })
+```
+
+You can also specify multiple key names as an array, which are tried in the order you provide them.
+This is most useful when you use the `recursive` option:
+
+```js
+var todos = [
+  {
+    id: 10,
+    description: 'Learn Things',
+    tags: [
+      { name: 'a', score: 15 }, { name: 'b', score: 83 }
+    ]
+  },
+  {
+    id: 20,
+    description: 'Do Things',
+    tags: [
+      { name: 'x', score: 3 }, { name: 'y', score: 9 }
+    ]
+  },
+]
+var todosMap = objectifyArray(todos, { by: ['id', 'name'], recursive: true })
+
+todosMap[10] //=> { id: 10, description: ..., tags: ... }
+todosMap[20] //=> { id: 10, description: ..., tags: ... }
+
+todosMap[10].tags.a.score //=> 15
+todosMap[10].tags.b.score //=> 83
 ```
 
-If your array contains nested arrays of objects, those are objectified too!
 For more extensive usage examples, see [test.js](test.js)
 
 ## API
 
 The module exports a single function:
 
-#### `objectifyArray(array[, keyName])`
+#### `objectifyArray(array[, options])`
 
-* `array` Array
-* `keyName` String (optional) - A property other than `id` or `name` to use for the object keys.
+* `array` Array - The array of objects to index.
+* `options` Object (optional) - `{ by: String | Array<String>, recursive: Boolean }`
```

Comment:
This looks good to me. Can you enumerate the options in a nested unordered list, and also document the default values for each option?

Reply:
👍